drivers/overwrite_queue.go

```diff
@@ -0,0 +1,119 @@
+package drivers
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/common"
+)
+
+const (
+	timeoutMultiplier = 1.5
+)
+
+type (
+	OvewriteQueue struct {
+		desc           string
+		name           string
+		session        OSSession
+		maxRetries     int
+		initialTimeout time.Duration
+		maxTimeout     time.Duration
+		queue          chan []byte
+		quit           chan struct{}
+	}
+)
+
+func NewOverwriteQueue(session OSSession, name, desc string, maxRetries int, initialTimeout, maxTimeout time.Duration) *OvewriteQueue {
+	oq := &OvewriteQueue{
+		desc:           desc,
+		name:           name,
+		maxRetries:     maxRetries,
+		session:        session,
+		initialTimeout: initialTimeout,
+		maxTimeout:     maxTimeout,
+		queue:          make(chan []byte, 32),
+		quit:           make(chan struct{}),
+	}
+	if maxRetries < 1 {
+		panic("maxRetries should be greater than zero")
+	}
+	go oq.workerLoop()
+	return oq
+}
+
+// Save queues data to be saved
+func (oq *OvewriteQueue) Save(data []byte) {
+	oq.queue <- data
+}
+
+// StopAfter stops reading loop after some time
+func (oq *OvewriteQueue) StopAfter(pause time.Duration) {
+	go func(p time.Duration) {
+		time.Sleep(p)
+		close(oq.quit)
+	}(pause)
+}
+
+// waitForQueueToClear used in tests
```

Comment:
wdyt about moving this to `drivers/overwrite_queue_test.go`?

Reply:
moved